src/IPGenerate/IPGenerator.cpp

```diff
@@ -427,18 +450,30 @@ bool IPGenerator::Generate() {
         jsonF << "{" << std::endl;
         for (auto param : inst->Parameters()) {
           std::string value;
-          switch (param.GetType()) {
-            case Value::Type::ParamIpVal:
-              value = param.GetSValue();
-              break;
-            case Value::Type::ParamString:
-              value = param.GetSValue();
-              break;
-            case Value::Type::ParamInt:
-              value = param.GetSValue();
-              break;
-            case Value::Type::ConstInt:
-              value = param.GetSValue();
+          // configure ip loses type info so we'll use the ip catalog's
```

Comment:
Can you we elaborate it with the reasons for the loss of type?

Reply:
updated